### ISSUE TITLE
Partial purge does not clear CPT archive when non-default cache files are used

### ISSUE THREAD

**Issue by Mai-Saad** [wp-media/wp-rocket#4508]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version => 3.10.3
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
Some Cached pages aren't cleared when editing post if Cookie GDPR is enabled

**To Reproduce**
Precondition:
- Cookie Notice & Compliance for GDPR / CCPA is enabled
- WooCommerce is enabled, at least one product is present, shop page is set

1. Visit the following URLs:
https://rocketlabsqa.ovh/product/wordpress-pennant/
https://rocketlabsqa.ovh/shop/
2. Accept cookie and refresh both pages
3. Edit the post and update:
https://rocketlabsqa.ovh/wp-admin/post.php?post=1000&action=edit
4. Cached files for the shop are not deleted

**Expected behavior**
All cached files are deleted

**Screenshots**
![cache](https://user-images.githubusercontent.com/76941962/142038377-71182306-2355-482a-a11f-288ae3ba1644.jpg)

**Additional context**
- If Cookie Notice & Compliance for GDPR / CCPA is deactivated, after step 3, all cached files will be deleted
- As per @piotrbak investigations
We're clearing only the .html  and .html_gzip ending files of the Archive page
When there's a dynamic cookie we will not clear the cache of specific archive pages
The problem is here: https://github.com/wp-media/wp-rocket/blob/9ba1546480baac14cd704622f8a575b47c0783a9/inc/common/purge.php#L72-L73
purge.php
It also seems that the pagination clearing is broken there, it tries to clear this https://rocketlabsqa.ovh/shop/index-httpspage instead of this https://rocketlabsqa.ovh/shop/page/2/index-https.html

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by webtrainingwheels** [comment]
This is not specific to Cookie GDPR but will happen whenever we create any non-default cache files. So it happens when WebP is active also, for example. @Mai-Saad maybe you could update the title to make it more general?

This is also related to: https://github.com/wp-media/wp-rocket/issues/2068

**Comment by Mai-Saad** [comment]
another scenario with WebP here https://github.com/wp-media/wp-rocket/issues/4617

**Comment by piotrbak** [comment]
There's no feedback from customers regarding this one. It's not going to be implemented in the near future. We're open to discuss and reopen the issue though.

**Comment by vellebelle** [comment]
I am having the same problem and was advised to install an extra plugin from WP Rocket to solve the issue. Why can't you add it to the core of WP Rocket as others are experiencing the same problem?

**Comment by worldwildweb** [comment]
related https://secure.helpscout.net/conversation/2819097199/534966
When a new custom post is created, if cached files exist, the archived desktop page is cleared, but the mobile page is not cleared.

**Comment by piotrbak** [comment]
**Acceptance Criteria**
When Partial Purge is happening, we need to make sure to clear cache of the archives:
1. mobile cache files 
2. webp cache files
3. dynamic cookies cache files
4. all combinations of the aboves

**Comment by Miraeld** [comment]
## Scope a solution

To fix the mentioned issue, we could modify the `$purge_urls[] =` part to include other files to be deleted. 
I also recommend optimising it.

``` php
if ( 'post' !== $post_type ) {
    $post_type_archive = get_post_type_archive_link( $post_type );

    if ( $post_type_archive ) {
        // Define base filename and SSL suffix
        $base_filename = 'index';
        $ssl_suffix = is_ssl() ? '-https' : '';

        $post_type_archive = trailingslashit( $post_type_archive );

        // Define base cache variations
        $filenames = [
            "{$base_filename}{$ssl_suffix}.html",
            "{$base_filename}{$ssl_suffix}.html_gzip",
            "{$base_filename}-mobile{$ssl_suffix}.html",
            "{$base_filename}-mobile{$ssl_suffix}.html_gzip",
            "{$base_filename}{$ssl_suffix}-true.html",
            "{$base_filename}{$ssl_suffix}-true.html_gzip",
            $GLOBALS['wp_rewrite']->pagination_base
        ];

        // Add full URLs to the purge array
        $purge_urls = array_map(fn($file) => $post_type_archive . $file, $filenames);
    }
}
```


This will include the `-true.html`, `-true.html_gzip` & `-mobile` files. 

This goes to replace the current code [here](https://github.com/wp-media/wp-rocket/blob/develop/inc/common/purge.php#L63-L77)

**Comment by wordpressfan** [comment]
Checking that now...

**Comment by wordpressfan** [comment]
@Miraeld what happens when we add the following line:
```
$purge_urls[]      = $post_type_archive;
```

directly after this line:

https://github.com/wp-media/wp-rocket/blob/5b286ce8c571285bd0924f8f223440e7f9f4c0b4/inc/common/purge.php#L75

can u plz check if this would work or not?

**Comment by Miraeld** [comment]
Umm, it seems to work, but, there is something I don't get. 
If I add a log to `$post_type_archive`, it is equal to: https://new.rocketlabsqa.ovh/shop-3 while I'm saving the page https://new.rocketlabsqa.ovh/product/resume-promo

Is it expected @wordpressfan ?

**Comment by wordpressfan** [comment]
Yes this is normal, if you checked the woocommerce **Settings > Products** the setting of `Shop page`, you will find the mentioned page `shop-3` with the ID `12034` for this specific site.

We use the function `get_post_type_archive_link` to get the custom post type archive's page which is set by woocommerce itself, exactly here:

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-post-types.php#L369

which is set here:

https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-post-types.php#L309

you can read more about has_archive argument here:

https://developer.wordpress.org/reference/functions/register_post_type/#parameters

does this make sense?
If that's working with u, do u mind updating the grooming to send it to Todo column?

Thanks Boss.

**Comment by wordpressfan** [comment]
## Grooming

We need to add the following line:

```
$purge_urls[]      = $post_type_archive;
```

directly after this line:

https://github.com/wp-media/wp-rocket/blob/5b286ce8c571285bd0924f8f223440e7f9f4c0b4/inc/common/purge.php#L75

### Effort:
[XS]
